src/api/submission/writer.py

```diff
@@ -77,27 +81,26 @@ def _create_sample_event(self, sample_event_data):
             raise ValidationError(serializer.errors)
 
         return serializer.save()
-    
+
     def get_sample_unit_method_id(self):
         return self.collect_record.data.get("sample_unit_method_id")
 
     def get_or_create_sample_event(self):
         sample_event_data = get_sample_event_data(self.collect_record)
-        try:
-            query_params = {
-                k: v for k, v in sample_event_data.items() if k not in ("id", "notes")
-            }
-            se = SampleEvent.objects.filter(**query_params)
-            if se.exists():
-                sample_event = se[0]
-                notes = sample_event_data.get("notes") or ""
-                if notes.strip():
-                    sample_event.notes += "\n\n{}".format(notes)
-                    sample_event.save()
-                return sample_event
-            return self._create_sample_event(sample_event_data)
-        except SampleEvent.DoesNotExist:
-            return self._create_sample_event(sample_event_data)
+        sample_event_data["id"] = uuid.uuid4()
+        serializer = self.validate_data(SampleEventSerializer, sample_event_data)
+        query_params = {
+            k: v for k, v in sample_event_data.items() if k not in ("id", "notes")
+        }
+        se = SampleEvent.objects.filter(**query_params)
+        if se.exists():
+            sample_event = se[0]
+            notes = sample_event_data.get("notes") or ""
+            if notes.strip():
+                sample_event.notes += "\n\n{}".format(notes)
+                sample_event.save()
+            return sample_event
+        return serializer.save()
```

Comment:
ok so with this refactor we should be able to delete the whole `_create_sample_event` method, right?